consensus/polybft/signer/signature.go

```diff
@@ -2,65 +2,52 @@ package bls
 
 import (
 	"errors"
+	"fmt"
 	"math/big"
 
-	"github.com/kilic/bn254"
+	bn256 "github.com/umbracle/go-eth-bn256"
 )
 
 // Signature represents bls signature which is point on the curve
 type Signature struct {
-	p *bn254.PointG1
+	g1 *bn256.G1
 }
 
 // Verify checks the BLS signature of the message against the public key of its signer
-func (s *Signature) Verify(publicKey *PublicKey, message []byte) bool {
-	e := bn254.NewEngine()
-	messagePoint, err := e.G1.HashToCurveFT(message, GetDomain())
-
+func (s *Signature) Verify(pub *PublicKey, message []byte) bool {
+	point, err := hashToPoint(message, domain)
 	if err != nil {
 		return false
 	}
 
-	e.AddPair(messagePoint, publicKey.p)
-	e.AddPairInv(s.p, e.G2.One())
-
-	return e.Check()
+	return bn256.PairingCheck([]*bn256.G1{s.g1, point}, []*bn256.G2{negG2Point, pub.g2})
 }
 
 // VerifyAggregated checks the BLS signature of the message against the aggregated public keys of its signers
 func (s *Signature) VerifyAggregated(publicKeys []*PublicKey, msg []byte) bool {
-	aggPubs := aggregatePublicKeys(publicKeys)
-
-	return s.Verify(aggPubs, msg)
+	return s.Verify(PublicKeys(publicKeys).Aggregate(), msg)
 }
 
-// Aggregate adds the given signatures
-func (s *Signature) Aggregate(next *Signature) *Signature {
-	g := bn254.NewG1()
-
-	newp := new(bn254.PointG1)
-	newp.Zero()
-
-	if s.p != nil {
-		if next.p != nil {
-			g.Add(newp, s.p, next.p)
-		} else {
-			newp.Set(s.p)
-		}
-	} else if next.p != nil {
-		newp.Set(next.p)
+// Marshal the signature to bytes.
+func (s *Signature) Marshal() ([]byte, error) {
+	if s.g1 == nil {
```

Comment:
Same as in PrivateKey.

Reply:
fixed https://github.com/0xPolygon/polygon-edge/pull/1161/commits/c39a660fed18b2128cee9bc0329b13d278d6c8b5